CromIAM/src/test/scala/cromiam/sam/SamClientSpec.scala

```diff
@@ -46,28 +49,37 @@ class SamClientSpec extends AsyncFlatSpec with Matchers with BeforeAndAfterAll
 
   behavior of "SamClient"
   it should "return true if user is whitelisted" in {
+    val samClient = new MockSamClient()
     samClient.isSubmitWhitelisted(authorizedUserWithCollection, emptyHttpRequest).map(v => assert(v))
       .asIo.unsafeToFuture()
   }
 
   it should "return false if user is not whitelisted" in {
+    val samClient = new MockSamClient()
     samClient.isSubmitWhitelisted(notWhitelistedUser, emptyHttpRequest).map(v => assert(!v))
       .asIo.unsafeToFuture()
   }
 
   it should "return sam errors while checking is whitelisted" in {
-    samClientWithError.isSubmitWhitelisted(notWhitelistedUser, emptyHttpRequest).value.unsafeToFuture() map {
-      _.left.value should be(expectedErrorResponse)
+    val samClient = new MockSamClient() {
+      override def isSubmitWhitelistedSam(user: User, cromiamRequest: HttpRequest): FailureResponseOrT[Boolean] = {
+        MockSamClient.returnResponse(expectedErrorResponse)
+      }
+    }
+    samClient.isSubmitWhitelisted(notWhitelistedUser, emptyHttpRequest).value.unsafeToFuture() map {
+      _ should be(Left(expectedErrorResponse))
     }
   }
 
   it should "eventually return the collection(s) of user" in {
+    val samClient = new MockSamClient()
     samClient.collectionsForUser(authorizedUserWithCollection, emptyHttpRequest).map(collectionList =>
-      assert(collectionList == samClient.userCollectionList)
+      assert(collectionList == MockSamClient.UserCollectionList)
     ).asIo.unsafeToFuture()
   }
 
   it should "fail if user doesn't have any collections" in {
+    val samClient = new MockSamClient()
```

Comment:
ToL (⚠️ "Semantics of 'mock' discussion" alert ⚠️): 

Personally I'd go with `TestableSamClient` rather than `MockSamClient` since this isn't really mocking the sam client, it's extending in such a way as to make it testable.

Reply:
Happy to rename this class, however the `Mock` prefix is an existing pattern throughout Cromiam specs though, and I'm not sure where to stop w/ the cleanup...

The `MockSamClient` is in a file called `MockClients`. In `MockClients` there's also a `MockCromwellClient`. Cromiam actually has two of these classes though, 
`cromiam.webservice.MockCromwellClient` and `cromiam.cromwell.CromwellClientSpec.MockCromwellClient`. And there's `MockCromwellApiClient` and `MockStatusService`.

In general, I'm more a fan of consistency (leave them all prefixed `Mock`, or change them _all_) but in this case I'm open to incremental progress?

Reply:
Not a strong opinion, just a TOL. Since it's not a new addition, I'd be happy to leave them as-is